### ISSUE TITLE
Rabbitmq fails to restart on first reconfigure

### ISSUE THREAD

**Issue by markan** [chef/chef-server#1690]
We have an intermittent failure on the first run of chef-server-ctl. Rerunning always seems to fix the problem. The failure is late enough in the first run that we're just running triggered restarts, and all configuration is done. 

### Chef Server Version
Chef Server 13.0.14

### Platform Details
Ubuntu 16.04, in vagrant/virtualbox (also seems to intermittently repro on AWS m4.large)

### Configuration
Standalone,  & Tiered, HA. New install or upgrade.

### Scenario:
Install chef server

### Steps to Reproduce:
dpkg -i chef-server-core_13.0.14-1_amd64.deb 
chef-server-ctl reconfigure --verbose  --chef-license accept

### Expected Result:
chef-server-ctl to succeed

### Actual Result:

      
---- Begin output of /opt/opscode/embedded/bin/sv restart /opt/opscode/service/rabbitmq ----

      
STDOUT: timeout: run: /opt/opscode/service/rabbitmq: (pid 7674) 68s

    
Full dump here (this is from a dev-vm but we can reproduce elsewhere)
https://gist.github.com/markan/ac0dbcf8ed9a9fab968868fbbff9bfcd

**Comment by markan** [comment]
So we've had trouble getting any insight into the underlying issue. The basic problem is that sv is timing out trying to stop rabbitmq; the process doesn't seem to be shutting down as expected.  This isn't totally unheard of, for example this seems to be related this. https://wiki.openstack.org/wiki/OpsGuide-Maintenance-RabbitMQ#RabbitMQ_service_hangs

It's been very difficult to capture into exactly what's happening in the process. I've tried some strategic strace usage during the timeout, but it's a bit of a timing trick to actually capture the command execution. If this drags on, probably should set up systemtap and erlang tracing and do some focussed work to figure out if the signal is being received and what's happening after. 

We have plans to eliminate rabbitmq (in favor of directly committing to Elasticsearch) in a near term release of chef-server, and so this may all be moot.

**Comment by markan** [comment]
This appears to be 'fixed' in 13.0.16. However, we're any closer to understanding why rabbitmq is refusing to stop

An interesting note is that in testing it appears that this bug has been with us for some time in one form or another, but has been elusive and difficult for us to pin down. Issues #1318 and perhaps #1643 may be related, and we should revisit them to see if they replicate in 13.0.16

One interesting fingerprint is log messages like this (from a 12.12.0 install!)
```
2019-06-28_22:35:37.63126 [error] gen_server <0.1412.0> terminated with reason: econnrefused in amqp_network_connection:do_connect/1 line 337
2019-06-28_22:35:37.63187 [error] CRASH REPORT Process <0.1412.0> with 0 neighbours exited with reason: econnrefused in gen_server:terminate/7 line 804
2019-06-28_22:35:37.63251 [error] Supervisor {<0.1410.0>,amqp_connection_sup} had child at module undefined at <0.1412.0> exit with reason econnrefused in context child_terminated
2019-06-28_22:35:37.63308 [error] Supervisor {<0.1410.0>,amqp_connection_sup} had child at module undefined at <0.1412.0> exit with reason reached_max_restart_intensity in context shutdown
2019-06-28_22:35:37.63374 [error] "Could not connect, scheduling reconnect.", error: {{error,{badmatch,{error,{auth_failure_likely,{econnrefused,{gen_server,call,[<0.1412.0>,connect,infinity]}}}}},[{bunny_util,connect,1,[{file,"/var/cache/omnibus/src/oc_erchef/_build/default/lib/gen_bunny/src/bunny_util.erl"},{line,191}]},{gen_bunny_mon,do_connect,3,[{file,"/var/cache/omnibus/src/oc_erchef/_build/default/lib/gen_bunny/src/gen_bunny_mon.erl"},{line,192}]},{gen_bunny_mon,handle_info,2,[{file,"/var/cache/omnibus/src/oc_erchef/_build/default/lib/gen_bunny/src/gen_bunny_mon.erl"},{line,134}]},{gen_server,try_dispatch,4,[{file,"gen_server.erl"},{line,593}]},{gen_server,handle_msg,5,[{file,"gen_server.erl"},{line,659}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,237}]}]},{connection_info,{network,{192,168,10,200},5672,{<<"actions">>,<<"a3e4ae0f2fee4bbfa7e2ba9d510cfedaa26ea318218fcef0c483e590c20d8973bcfce6912d6ce6bd046b14e0fb3ad6416c8e">>},<<"/analytics">>}}}
2019-06-28_22:35:38.62605 [error] Could not start the network driver: econnrefused
2019-06-28_22:35:38.62606 
2019-06-28_22:35:38.63228 [error] gen_server <0.1415.0> terminated with reason: econnrefused in amqp_network_connection:do_connect/1 line 337
2019-06-28_22:35:38.63296 [error] CRASH REPORT Process <0.1415.0> with 0 neighbours exited with reason: econnrefused in gen_server:terminate/7 line 804
2019-06-28_22:35:38.63363 [error] Supervisor {<0.1413.0>,amqp_connection_sup} had child at module undefined at <0.1415.0> exit with reason econnrefused in context child_terminated
2019-06-28_22:35:38.63428 [error] Supervisor {<0.1413.0>,amqp_connection_sup} had child at module undefined at <0.1415.0> exit with reason reached_max_restart_intensity in context shutdown
2019-06-28_22:35:38.63498 [error] "Could not connect, scheduling reconnect.", error: {{error,{badmatch,{error,{auth_failure_likely,{econnrefused,{gen_server,call,[<0.1415.0>,connect,infinity]}}}}},[{bunny_util,connect,1,[{file,"/var/cache/omnibus/src/oc_erchef/_build/default/lib/gen_bunny/src/bunny_util.erl"},{line,191}]},{gen_bunny_mon,do_connect,3,[{file,"/var/cache/omnibus/src/oc_erchef/_build/default/lib/gen_bunny/src/gen_bunny_mon.erl"},{line,192}]},{gen_bunny_mon,handle_info,2,[{file,"/var/cache/omnibus/src/oc_erchef/_build/default/lib/gen_bunny/src/gen_bunny_mon.erl"},{line,134}]},{gen_server,try_dispatch,4,[{file,"gen_server.erl"},{line,593}]},{gen_server,handle_msg,5,[{file,"gen_server.erl"},{line,659}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,237}]}]},{connection_info,{network,{192,168,10,200},5672,{<<"actions">>,<<"a3e4ae0f2fee4bbfa7e2ba9d510cfedaa26ea318218fcef0c483e590c20d8973bcfce6912d6ce6bd046b14e0fb3ad6416c8e">>},<<"/analytics">>}}}
2019-06-28_22:35:39.63984 [error] Could not start the network driver: econnrefused
2019-06-28_22:35:39.63985 
2019-06-28_22:35:39.64054 [error] gen_server <0.1418.0> terminated with reason: econnrefused in amqp_network_connection:do_connect/1 line 337
2019-06-28_22:35:39.64117 [error] CRASH REPORT Process <0.1418.0> with 0 neighbours exited with reason: econnrefused in gen_server:terminate/7 line 804
2019-06-28_22:35:39.64186 [error] Supervisor {<0.1416.0>,amqp_connection_sup} had child at module undefined at <0.1418.0> exit with reason econnrefused in context child_terminated
2019-06-28_22:35:39.64252 [error] Supervisor {<0.1416.0>,amqp_connection_sup} had child at module undefined at <0.1418.0> exit with reason reached_max_restart_intensity in context shutdown
2019-06-28_22:35:39.64311 [error] "Could not connect, scheduling reconnect.", error: {{error,{badmatch,{error,{auth_failure_likely,{econnrefused,{gen_server,call,[<0.1418.0>,connect,infinity]}}}}},[{bunny_util,connect,1,[{file,"/var/cache/omnibus/src/oc_erchef/_build/default/lib/gen_bunny/src/bunny_util.erl"},{line,191}]},{gen_bunny_mon,do_connect,3,[{file,"/var/cache/omnibus/src/oc_erchef/_build/default/lib/gen_bunny/src/gen_bunny_mon.erl"},{line,192}]},{gen_bunny_mon,handle_info,2,[{file,"/var/cache/omnibus/src/oc_erchef/_build/default/lib/gen_bunny/src/gen_bunny_mon.erl"},{line,134}]},{gen_server,try_dispatch,4,[{file,"gen_server.erl"},{line,593}]},{gen_server,handle_msg,5,[{file,"gen_server.erl"},{line,659}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,237}]}]},{connection_info,{network,{192,168,10,200},5672,{<<"actions">>,<<"a3e4ae0f2fee4bbfa7e2ba9d510cfedaa26ea318218fcef0c483e590c20d8973bcfce6912d6ce6bd046b14e0fb3ad6416c8e">>},<<"/analytics">>}}}
2019-06-28_22:35:40.66283 [error] Could not start the network driver: econnrefused
2019-06-28_22:35:40.66284 
2019-06-28_22:35:40.66351 [error] gen_server <0.1421.0> terminated with reason: econnrefused in amqp_network_connection:do_connect/1 line 337
```
`chef-server-ctl stop` then `chef-server-ctl start` seems to fix this; sometime a hard kill is required to restart rabbitmq. 

The error above seems to be because the actions queue is missing, which is created via rabbitmqctl. 

The change we made in #1691 (in 13.0.16) was to ensure that rabbitmq was restarted after we configured rabbitmqctl; otherwise rabbitmqctl doesn't work. I strongly suspect that the 12.12.0 issue above is a similar thing; we'd configured rabbitmqctl but not restarted rabbitmq; once we restarted rabbitmq everything worked again. 

So it looks like we've had this bug for ages, but upgrading runit/chef-client has taken an intermittent, hard to replicate bug and made it much more frequent.

**Comment by stevendanna** [comment]
RabbitMQ is no longer in the Chef Infra Server.
